src/unix/fs.c

```diff
@@ -217,6 +217,82 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static int uv__fs_mkstemp(uv_fs_t* req) {
+  int r;
+#ifdef UV__O_CLOEXEC
+  int (*mkostemp_function)(char*, int);
+  static int no_cloexec_support;
+#endif
+  char *path;
+  int path_length;
+
+  path = (char*) req->path;
+  path_length = strlen(path);
+
+  /* EINVAL can be returned for 2 reasons:
+       1. The template's last 6 characters were not XXXXXX
+       2. open() didn't support O_CLOEXEC
+     We want to avoid going to the fallback path in case
+     of 1., so it's manually checked before. */
+  if (path_length < 6 ||
+      strcmp(path + (path_length - 6), "XXXXXX")) {
+    errno = EINVAL;
+    return -1;
+  }
+
+#ifdef UV__O_CLOEXEC
+  if (no_cloexec_support == 0) {
+    *(int**)(&mkostemp_function) = dlsym(RTLD_DEFAULT, "mkostemp");
```

Comment:
I guess we can just ignore `mkostemp` if we can't reliably load it at runtime. @bnoordhuis WDYT?